flask_rebar/validation.py

```diff
@@ -17,12 +18,14 @@
 FilterResult = namedtuple("FilterResult", "loadable, dump_only")
 
 
-def filter_dump_only(schema, data):
+def filter_dump_only(
+    schema: Schema, data: Union[Mapping[str, Any], List[Mapping[str, Any]]]
+) -> FilterResult:
     """
     Return a filtered copy of data in which any items matching a "dump_only" field are separated
     :param schema: Instance of a Schema class
-    :param data: Dict or collection of dicts with data
-    :return: Union[FilterResult, list[FilterResult]]
+    :param data: Mapping or List of Mappings with data
```

Comment:
think whether or not we use `Mapping / Dict` or `List/Collection` is actually dictated by the type hints in marshmallow